native/cocos/bindings/manual/jsb_global.cpp

```diff
@@ -659,7 +662,69 @@ static bool js_loadImage(se::State &s) { // NOLINT
     return false;
 }
 SE_BIND_FUNC(js_loadImage)
+//pixels(RGBA), width, height, fullFilePath(*.png/*.jpg)
+static bool js_saveImageData(se::State& s) // NOLINT
+{
+    const auto& args = s.args();
+    size_t argc = args.size();
+    CC_UNUSED bool ok = true;
+    if (argc == 4 || argc == 5) {
+        auto *uint8ArrayObj = args[0].toObject();
+        uint8ArrayObj->root();
+        uint8ArrayObj->incRef();
+        uint8_t *uint8ArrayData = nullptr;
+        size_t length = 0;
+        
+        uint8ArrayObj->getTypedArrayData(&uint8ArrayData, &length);
+        uint32_t width;
+        uint32_t height;
+        ok &= sevalue_to_native(args[1], &width);
+        ok &= sevalue_to_native(args[2], &height);
+
+        std::string filePath;
+        ok &= sevalue_to_native(args[3], &filePath);
+        SE_PRECONDITION2(ok, false, "js_saveImageData : Error processing arguments");
+        
+        se::Value callbackVal = argc == 5 ? args[4] : se::Value::Null;
+        se::Object *callbackObj;
+        if (!callbackVal.isNull()) {
+            CC_ASSERT(callbackVal.isObject());
+            CC_ASSERT(callbackVal.toObject()->isFunction());
+            callbackObj = callbackVal.toObject();
+            s.thisObject()->attachObject(callbackObj);
+            callbackObj->root();
+            callbackObj->incRef();
+        }
 
+        gThreadPool->pushTask([=](int /*tid*/) {
+            // isToRGB = false, to keep alpha channel
+            auto *img = ccnew Image();
+            img->initWithRawData(uint8ArrayData, length, width, height, 32 /*Unused*/);
```

Comment:
Consider to put the logic of new and initialize Image in logic thread to avoid saving damaged data.
In that case, uint8ArrayObj doesn't need to be rooted too.

Reply:
1